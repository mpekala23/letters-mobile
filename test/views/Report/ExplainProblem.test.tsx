import React from 'react';
import { ExplainProblemScreen } from '@views';
import { render, fireEvent, toJSON } from '@testing-library/react-native';

const setup = () => {
  const navigation = { navigate: jest.fn() };
  return {
    navigation,
    ...render(<ExplainProblemScreen navigation={navigation} />),
  };
};

describe('Explain Problem screen', () => {
  it('should match snapshot', () => {
    const { container } = setup();
    const tree = toJSON(container);
    expect(tree).toMatchSnapshot();
  });
  it('should navigate to thanks screen when report the problem is pressed', async () => {
    const { navigation, getByText } = setup();
    const reportButton = getByText('Report the problem');
    fireEvent.press(reportButton);
    expect(navigation.navigate).toHaveBeenCalledWith('Thanks');
  });
});
