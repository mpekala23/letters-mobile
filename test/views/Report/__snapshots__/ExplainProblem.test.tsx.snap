// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Explain Problem screen should match snapshot 1`] = `
<View
  collapsable={true}
  pointerEvents="box-none"
  style={
    Object {
      "flex": 1,
    }
  }
>
  <TouchableOpacity
    activeOpacity={1}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "white",
          "flex": 1,
          "paddingHorizontal": 30,
          "paddingTop": 40,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "Poppins-Bold",
            },
            Object {
              "fontSize": 23,
              "textAlign": "center",
            },
          ]
        }
      >
        Report problem
      </Text>
      <View
        style={
          Object {
            "alignItems": "flex-start",
            "backgroundColor": "#F6F6F6",
            "borderRadius": 4,
            "height": "50%",
            "justifyContent": "flex-start",
            "marginTop": 10,
            "padding": 10,
            "width": "100%",
          }
        }
      >
        <TextInput
          allowFontScaling={true}
          multiline={true}
          numberOfLines={10}
          placeholder="Type something"
          placeholderTextColor="grey"
          rejectResponderTermination={true}
          style={
            Array [
              Object {
                "fontFamily": "Poppins-Regular",
              },
              Object {
                "height": "100%",
                "justifyContent": "flex-start",
                "width": "100%",
              },
            ]
          }
          underlineColorAndroid="transparent"
        />
      </View>
      <TouchableOpacity
        activeOpacity={0.7}
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "#FF7171",
              "borderRadius": 15,
              "flexDirection": "row",
              "justifyContent": "center",
              "marginVertical": 5,
              "paddingHorizontal": 14,
            },
            Object {},
            Object {
              "marginTop": 30,
              "width": "100%",
            },
            Object {},
            Object {
              "elevation": 4,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 2,
                "width": 0,
              },
              "shadowOpacity": 0.23,
              "shadowRadius": 2.62,
            },
          ]
        }
        testID="clickable"
      >
        <Text
          style={
            Array [
              Object {
                "fontFamily": "Poppins-Medium",
              },
              Object {
                "color": "white",
                "fontSize": 20,
                "fontWeight": "700",
                "padding": 6,
              },
              Object {},
              Object {},
              Object {},
            ]
          }
        >
          Report the problem
        </Text>
      </TouchableOpacity>
    </View>
  </TouchableOpacity>
</View>
`;
